rust/rope/src/engine.rs

```diff
@@ -222,6 +234,18 @@ impl Engine {
         self.rev_id_counter += 1;
     }
 
+    pub fn compare(&self, base_rev: usize, other_rev: usize) -> bool {
```

Comment:
agreed, "compare" could be lots of things. I'd go for "is_equivalent_revision" to go with [Rust naming style](https://aturon.github.io/style/naming.html#predicates).